aries_vcx/src/protocols/did_exchange/state_machine/mod.rs

```diff
@@ -110,3 +119,67 @@ impl<I, S> DidExchange<I, S> {
         }
     }
 }
+
+impl<I, S> DidExchange<I, S> {
+    pub fn our_did_doc(&self) -> &DidDocumentSov {
+        &self.our_did_document
+    }
+
+    pub fn their_did_doc(&self) -> &DidDocumentSov {
+        &self.their_did_document
+    }
+
+    pub fn get_endpoint(&self) -> Option<Url> {
+        let did_document = self.their_did_doc();
+        // todo: no unwrap pls
+        let service = did_document.service().first()?;
+        // todo: will get cleaned up after service is de-generified
+        Some(match service {
+            ServiceSov::Legacy(d) => d.service_endpoint().into(),
+            ServiceSov::AIP1(d) => d.service_endpoint().into(),
+            ServiceSov::DIDCommV1(d) => d.service_endpoint().into(),
+            ServiceSov::DIDCommV2(d) => d.service_endpoint().into(),
+        })
+    }
+    pub async fn encrypt_message(
```

Comment:
I don't think `encrypt_message` and `send_message` belong under didx state machine at all

Reply:
removed